src/zen_ecdh.c

```diff
@@ -352,9 +352,38 @@ static int ecdh_encrypt(lua_State *L) {
 	// output is padded to next word
 	octet *out = o_new(L, in->len+0x0f); SAFE(out);
 	AES_CBC_IV0_ENCRYPT(k,in,out);
+
 	return 1;
 }
 
+/**
+   AES-GCM encrypt with Additional Data (AEAD)
+   encrypts and authenticate a plaintext to a ciphtertext. IEEE P802.1
+
+   @param key AES key octet
+   @param message input text in an octet
+   @param iv initialization vector
+   @param header the additional data
+   @function keyring:aead_encrypt(key, message, iv, h)
+   @treturn[1] octet containing the output ciphertext
+   @treturn[1] octet containing the authentication tag (checksum)
+*/
+
+static int ecdh_aead_encrypt(lua_State *L) {
+	HERE();
+	ecdh *e = ecdh_arg(L, 1); SAFE(e);
+	octet *k = o_arg(L, 2); SAFE(k);
+	octet *in = o_arg(L, 3); SAFE(in);
+	octet *iv = o_arg(L, 4); SAFE(iv);
+	octet *h = o_arg(L, 5); SAFE(h);
+
+	// output is padded to next word
+	octet *out = o_new(L, in->len+16); SAFE(out);
+	octet *t = o_new(L, 16); SAFE (t);
```

Comment:
is this size 16 including a small boundary?
Elsewhere in Zenroom I use 2 bytes.
Padding may it may solve the current segfault in OSX.

Reply:
Usually the tag size is 16 bytes. I can try adding a small boundary as you suggested.

Reply:
it wasn't the boundary in this case. will merge as tests are passing.